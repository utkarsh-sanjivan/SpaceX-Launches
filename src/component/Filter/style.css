.filters-conatainer {
    padding: 10px;
    margin: 10px;
    background-color: #fff;
}

.filters-title {
    font-size: 16px;
    font-weight: 600;
}

.filter-type-text {
    font-size: 12px;
    line-height: 18px;
    margin: 5px 20% 10px 20%;
    width: 60%;
    text-align: center;
    border-bottom: 1px black solid;
}

.filter-type-buttons {
    display: grid;
    grid-template-columns: auto auto;
    grid-gap: 10px 30px;
    width: 75%;
    margin: 0 12.5% 20px 12.5%;
}

.filter-button-div {
    width: 100%;
    height: 100%;
}

@media only screen and (max-width: 242px) {
    .filter-type-buttons {
        grid-template-columns: auto;
    }
}