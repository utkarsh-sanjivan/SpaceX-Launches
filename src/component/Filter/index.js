import React from 'react';
import SpaceXButton from './../SpaceXButton/index';
import './style.css';

const Filter = props => (
    <div className="filters-conatainer">
        <div className="filters-title">
            Filters
        </div>
        <div className="filter-type-text">Launch Year</div>
        <div style={{ gridTemplateRows: '30px 30px 30px 30px 30px 30px 30px 30px' }} className="filter-type-buttons">
            {Array(15).fill(null).map((__, index) => 
                <div className="filter-button-div">
                    <SpaceXButton 
                        isSelected={props.filterObj.launch_year === (index+2006).toString()}
                        onClick={() => props.filterClicked('launch_year', (index+2006).toString() )}
                    >
                        {index+2006}
                    </SpaceXButton>
                </div>
            )}
        </div>
        <div className="filter-type-text">Successful Launch</div>
        <div style={{ gridTemplateRows: '30px' }} className="filter-type-buttons">
            {[{ text: 'True', checkText: 'true', filterType: 'launch_success' }, { text: 'False', checkText: 'false', filterType: 'launch_success' }].map(value => 
                <div className="filter-button-div">
                    <SpaceXButton 
                        isSelected={props.filterObj.launch_success === value.checkText}
                        onClick={() => props.filterClicked(value.filterType, value.checkText)}
                    >
                        {value.text}
                    </SpaceXButton>
                </div>
            )}
        </div>
        <div className="filter-type-text">Successful Landing</div>
        <div style={{ gridTemplateRows: '30px' }} className="filter-type-buttons">
            {[{ text: 'True', checkText: 'true', filterType: 'land_success' }, { text: 'False', checkText: 'false', filterType: 'land_success' }].map(value => 
                <div className="filter-button-div">
                    <SpaceXButton 
                        isSelected={props.filterObj.land_success === value.checkText}
                        onClick={() => props.filterClicked(value.filterType, value.checkText)}
                    >
                        {value.text}
                    </SpaceXButton>
                </div>
            )}
        </div>
    </div>
)

export default Filter;