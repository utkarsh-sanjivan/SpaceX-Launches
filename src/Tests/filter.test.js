import React from 'react';
import { create } from "react-test-renderer";
import Filter from './../component/Filter/index';

test('Test SpaceX FIlter Component', () => {
  const component = create(<Filter 
        filterObj={{
            launch_year: '',
            launch_success: '',
            land_success: '',
        }}
        filterClicked={() => {}}
    />);
  expect(component.toJSON()).toMatchSnapshot()
});