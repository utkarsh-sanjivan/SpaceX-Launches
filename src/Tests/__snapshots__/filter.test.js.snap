// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SpaceX FIlter Component 1`] = `
<div
  className="filters-conatainer"
>
  <div
    className="filters-title"
  >
    Filters
  </div>
  <div
    className="filter-type-text"
  >
    Launch Year
  </div>
  <div
    className="filter-type-buttons"
    style={
      Object {
        "gridTemplateRows": "30px 30px 30px 30px 30px 30px 30px 30px",
      }
    }
  >
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2006
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2007
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2008
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2009
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2010
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2011
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2012
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2013
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2014
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2015
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2016
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2017
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2018
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2019
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        2020
      </button>
    </div>
  </div>
  <div
    className="filter-type-text"
  >
    Successful Launch
  </div>
  <div
    className="filter-type-buttons"
    style={
      Object {
        "gridTemplateRows": "30px",
      }
    }
  >
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          True
        </span>
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          False
        </span>
      </button>
    </div>
  </div>
  <div
    className="filter-type-text"
  >
    Successful Landing
  </div>
  <div
    className="filter-type-buttons"
    style={
      Object {
        "gridTemplateRows": "30px",
      }
    }
  >
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          True
        </span>
      </button>
    </div>
    <div
      className="filter-button-div"
    >
      <button
        className="ant-btn unselected-color"
        isSelected={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          False
        </span>
      </button>
    </div>
  </div>
</div>
`;
